src/main/java/com/segment/analytics/android/integrations/google/analytics/GoogleAnalyticsIntegration.java

```diff
@@ -333,46 +357,60 @@ private static int extractNumber(String text, int start) {
   }
 
   /** Send a product event. */
-  void sendProductEvent(String event, String category, Properties properties) {
+  void sendProductEvent(String event, String category, Properties eventProperties) {
     if (!PRODUCT_EVENT_NAME_PATTERN.matcher(event).matches()) {
       return;
     }
 
-    com.google.android.gms.analytics.ecommerce.Product product =
-            new com.google.android.gms.analytics.ecommerce.Product()
-        .setId(properties.productId())
-        .setName(properties.name())
-        .setCategory(isNullOrEmpty(category) ? DEFAULT_CATEGORY : category)
-        .setPrice(properties.price())
-        .setQuantity(properties.getInt(QUANTITY_KEY, 0));
+    List<Product> products = eventProperties.products();
+    if (isNullOrEmpty(products)) {
```

Comment:
Is this for single product events, ex. Product Addded product information sits at the top level as properties

Reply:
Yep, this is mostly to preserve backward compatibility, e.g. if you sent an event without any `products` array.